include/deal.II/base/type_traits.h

```diff
@@ -0,0 +1,126 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__type_traits_h
+#define dealii__type_traits_h
+
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/la_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace std_cxx11
+{
+
+  /**
+  * A true_type similar to c++11 std::true_type
+  *
+  * @author Uwe Köcher, 2017
+  */
+  struct true_type
+  {
+    static const bool value = true;
```

Comment:
I think these classes should go into a file `include/deal.II/base/std_cxx11/type_traits.h` -- so we know that we can throw them away after the switch to C++11.